### ISSUE TITLE
Feature Request: tag to prevent spawning and deny NPC passage

### ISSUE THREAD

**Issue by patrickheney** [MeridiusIX/ModularEncountersSpawner#28]
[SpawnBlockedByBlock:true]
// Whether or not to enable NPC spawn "blocking"

[SpawnBlockingBlockId:Antenna_ElectricFence]
// The name of the antenna block that is used to block the NPC spawn

[SpawnBlockEffectiveness:5]
// The percentage of the antenna's broadcast range that block NPC spawns.

This prevents spawns from appearing within 5% of the broadcast range of any Antenna block named "Antenna_ElectricFence." The NPCs are still be spawned, they are just located outside the blocked area.

[SpawnBlockingDeniesEntry:true]
This causes NPC's to attempt to leave the area described above (5% broadcast range of the antenna).

Uses:
Virtual fences to keep out spiders and wolves.
A "keep away" signal effective against some NPCs (for whatever narrative reason fits -- maybe it's ECM or something).

This has a nice balancing drawback that the bigger the protected area, the larger the broadcast radius announcing your presence to all the other NPCs and players.

**Comment by MeridiusIX** [comment]
I'm assuming you intend this for NPC bots right (creatures, wolves, spiders, etc)? Some of it might be possible, but other elements wouldn't be (such as denies entry, we have little to no control over how creatures behave once spawned). I'll consider the idea though, since I'd first need a way to track all the blocks in a performant manner.

**Comment by patrickheney** [comment]
Yes, that was the intent -- for NPC bots (creatures, wolves, spiders).
